lib/Onyx.js

```diff
@@ -926,19 +934,12 @@ function hasPendingMergeForKey(key) {
  * @private
  * @param {String} key
  * @param {*} data
- *
  * @returns {*}
  */
 function applyMerge(key, data) {
     const mergeValues = mergeQueue[key];
-    if (_.isArray(data) || _.every(mergeValues, _.isArray)) {
-        // Array values will always just concatenate
-        // more items onto the end of the array
-        return _.reduce(mergeValues, (modifiedData, mergeValue) => [
-            ...modifiedData,
-            ...mergeValue,
-        ], data || []);
-    }
+
+    if (_.isArray(data) || _.every(mergeValues, _.isArray)) { return _.last(mergeValues); }
```

Comment:
It'll probably help if we leave a comment about why we do this

Reply:
We also have to be careful as this does change some expectations about how we handle array merging. But last time I looked we got rid of most logic that concatenated arrays.

This comment needs to be updated as well:

https://github.com/Expensify/react-native-onyx/blob/e5ac3187565e4e2fad0504da6a121d8419cb01dc/lib/Onyx.js#L966-L969

Reply:
Updated the comment, though one of you should probably look over it, maybe the explanation can be formed more easily..